### PR TITLE
Us77969 sync enrollments

### PR DESCRIPTION
**Pull request by apalaniuk**
This isn't very Polymer behavior-y, so if it should be moved to its own repo or somewhere better, I don't want to clutter things (more following the pattern that the `Id` namespace also exists here, but this is a bit larger..)

Implements some basic publish/subscribe functionality to allow decoupled message passing between unrelated components. Not intended to implement anything too complex for the moment (eg. actual channel hierarchies).

### CODE REVIEW

File: d2l-publish-subscribe.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			}
+		},
+
+		unsubscribe: function(channel, subscriber) {
+
+			if (!this._channels[channel]) {
+				return;
+			}
+
+			if (this._channels.hasOwnProperty(channel)) {
```
Comment (dbatiste) [inline d2l-publish-subscribe.html]: Does this check essentially duplicate the check on line 20?

File: d2l-publish-subscribe.html [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+					if (this._channels[channel].length === 0) {
+						delete this._channels[channel];
+					}
+				}
+			}
+		},
+
+		publish: function(channel, message) {
+
+			if (this._channels[channel]) {
```
Comment (dbatiste) [inline d2l-publish-subscribe.html]: Formatting preference:
```
if (!this._channels[channel]) {
   return;
}

this._channels[channel].forEach(function(subscriber) {
   ...
});
```

File: d2l-publish-subscribe.html (hunk 1 above)
Comment (apalaniuk) [reply d2l-publish-subscribe.html]: So it does.

**Comment by dbatiste** [comment]
Seems legit.  The analogous of this in MVC are the custom event handlers.

My understanding is that this will be used to enable trigger the nav to re-graft the course selector when courses are pinned in my-courses widget so that the state is reflected correctly in the grafted menu.  Pinning can happen in either place, so I suspect each would be a subscriber and a publisher?    

@dlockhart : what are your thoughts on this?

**Comment by dlockhart** [comment]
I was hoping that the nav-re-graft wouldn't be necessary if we took another approach (have the existing course selector listen for the events and change its own icons). So not sure if this is still necessary if that approach changes.

**Comment by apalaniuk** [comment]
@dlockhart @dbatiste I definitely agree that avoiding re-grafts is a better idea and am looking at that - I think it could be done with or without this as a mediator, though the communicating components being unrelated made this seem like the natural pattern. Will give straight eventing a shot.

**Pull request closed without merging**